test/e2e_scheduler/waitforfirstconsumer.go

```diff
@@ -0,0 +1,90 @@
+package e2escheduler
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/loft-sh/vcluster/test/framework"
+	"github.com/onsi/ginkgo"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = ginkgo.Describe("Schedule a Statefulset with WaitForFirstConsumer PVCs", func() {
+	// the default kind storageclass is WaitForFirtConsumer
+	f := framework.DefaultFramework
+	// create statefulset with pvc
+	workload := &appsv1.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-statefulset",
+			Namespace: "default",
+		},
+		Spec: appsv1.StatefulSetSpec{
+			Replicas: int32Ref(1),
+			VolumeClaimTemplates: []corev1.PersistentVolumeClaim{
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "data"},
+					Spec: corev1.PersistentVolumeClaimSpec{
+						AccessModes: []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce},
+						Resources: corev1.ResourceRequirements{
+							Requests: corev1.ResourceList{
+								corev1.ResourceStorage: resource.MustParse("5Gi"),
+							},
+						},
+					},
+				},
+			},
+			Selector: &metav1.LabelSelector{MatchLabels: map[string]string{"app": "test"}},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{"app": "test"},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "nginx",
+							Image: "nginx",
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      "data",
+									MountPath: "/data",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	ginkgo.It("Wait for Statefulset to become ready", func() {
+		// TODO: initialize a controller-runtime client in the framework
```

Comment:
There is `f.VclusterClient` which you could use here.

Reply:
I think it's a subconscious leftover from porting over controller code that used uncached clients to cached clients. Yeah, should be totally fine to use the client-go client in e2e

Reply:
After having worked with the regular client-go client a bit more while I was away from this PR. I now remember why I like the controller-runtime client so much more. It's generic and much more ergonomic!

Reply:
I like client-go in the tests because then tests are a bit easier to read, they look a bit closer to the "BDD test cases".
But this is pretty subjective, so no problem with keeping it as is :)